src/Concordium/Client/Output.hs

```diff
@@ -699,6 +728,26 @@ showEvent verbose = \case
     showDelegationTarget Types.DelegatePassive = "Passive delegation"
     showDelegationTarget (Types.DelegateToBaker bid) = "Baker ID " ++ show bid
 
+    showLoggedEvents :: [Wasm.ContractEvent] -> String
+    showLoggedEvents [] = "No contract events were emitted "
+    showLoggedEvents evs = [i|#{length evs} contract events were emitted|]
+      <> if isNothing stM
+         then [i| but no event schema was provided nor found in the contract module. |]
+         else [i|, of which #{length $ filter isJust $ map snd loggedEvents} were succesfully parsed using schema. |]
+      <> [i|Got:\n|]
+      <> intercalate "\n" [ if isJust json
+                            then [i|Event(parsed):\n#{indentBy 4 $ fromMaybe' json}|]
+                            else [i|Event(raw): #{hex}\n|]
+                            | (hex, json) <- loggedEvents ]
+        where
+          fromMaybe' :: Maybe AE.Value -> String
+          fromMaybe' = \case Nothing -> "Unable to decode."; Just j -> showSortedPrettyJSON j
+          toHex :: Wasm.ContractEvent  -> String
+          toHex (Wasm.ContractEvent bs) = show . BSB.toLazyByteString . BSB.byteStringHex $ SE.runPut $ SE.putShortByteString bs
+          toJSON' :: Wasm.ContractEvent -> Maybe AE.Value
+          toJSON' (Wasm.ContractEvent bs) =
+            stM >>= \st' -> (preview _Right . PA.decodeParameter st' . SE.runPut . SE.putShortByteString) bs
```

Comment:
Can you honestly say that in 1month this code will be readable? 

Please just do a simple thing. A line or two more is better than weird sequences of function compositions with hard to understand intermediate results.
```
-- Attempt to decode the contract event if the schema is provided.
-- If there is no schema, or decoding fails @Nothing@ is returned.
toJSON' (Wasm.ContractEvent bs) = do
    st' <- stM
    case PA.decodeParameter (BSS.fromShort st') of
        Left _ -> Nothing
        Right x -> Just x
```